common/network-common/src/main/java/org/apache/spark/network/client/TransportClientFactory.java

```diff
@@ -121,18 +126,22 @@ public MetricSet getAllMetrics() {
   /**
    * Create a {@link TransportClient} connecting to the given remote host / port.
    *
-   * We maintains an array of clients (size determined by spark.shuffle.io.numConnectionsPerPeer)
+   * We maintain an array of clients (size determined by spark.shuffle.io.numConnectionsPerPeer)
    * and randomly picks one to use. If no client was previously created in the randomly selected
    * spot, this function creates a new client and places it there.
    *
+   * We also maintain a last connection failed time of these clients and a fast fail time window
+   * based on io retry wait time. If this connection request can be retried and the last connection
+   * failed time of these clients in the fast fail time window, fail this connection directly.
+   *
    * Prior to the creation of a new TransportClient, we will execute all
    * {@link TransportClientBootstrap}s that are registered with this factory.
    *
    * This blocks until a connection is successfully established and fully bootstrapped.
    *
    * Concurrency: This method is safe to call from multiple threads.
    */
-  public TransportClient createClient(String remoteHost, int remotePort)
+  public TransportClient createClient(String remoteHost, int remotePort, boolean withRetry)
```

Comment:
instead of calling this withRetry - which reading without context makes me think this is going to retry, perhaps we should rename this and call it failFast or something similar failFastOn.... I'm trying to come up with better name but can't think of anything.
also I realize this didn't have java doc before, but I think we need it for this parameter so can you please add in the java doc for the parameters and describe the new parameter. Make sure that is sufficient or add in a comment by line 204 as well.

Reply:
thanks, have named this variable to `fastFail` and add java doc and comments.